### PR TITLE
VxDesign: Load and convert MS SEMS elections

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

Task: https://github.com/votingworks/vxsuite/issues/7312

Adds the ability to load MS SEMS election files and convert them into a VX election via the "Load Election" button.

The conversion code is based on our historical [SEMS converter code](https://github.com/votingworks/vxsuite/blob/m16-dev-branch/services/converter-ms-sems/converter/SEMSinput.py). The nitty gritty details are explained in the code, but the high level is that the user needs to upload two CSV files that are exported from SEMS, one that has most of the election data, and the other that has a mapping of candidate IDs.

One important detail is that we need to save the SEMS IDs for election entities so that we can later use them to convert election results for import back into SEMS. We do this the same way that we did this previously - by using these SEMS IDs as the internal IDs for each election entity. To ensure they are unique and can serve as db primary keys, we prefix them with the election ID (which is a normal VX ID generated by VxDesign).

This functionality is currently controlled by a new feature flag that is only enabled for the VotingWorks org. I plan to figure out how to expose this for MS orgs (of which there are currently none) in a future PR.

## Demo Video or Screenshot

https://github.com/user-attachments/assets/d636468f-ebf7-4672-bed6-30e7559a35ee

## Testing Plan
Mostly automated testing with sample files. It's a bit hard to ensure that the resulting elections are correct (beyond basic validation). I did some manual inspection of the resulting elections (which are snapshotted). I assume we'll figure out some of the edge cases as we use the feature more, but I'm open to other ideas for testing this ahead of time.

Note that there are still some backend integration tests and frontend tests that I have yet to write.
## Checklist

- [x] I have prefixed my PR title with "VxDesign: ", "VxPollBook: ", or "HWTA: " if my change is specific to one of those products.
- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate for any new user actions.
- [x] I have added the "user_facing_change" label to this PR, if relevant, to automate an announcement in #machine-product-updates.

### CODE REVIEW

File: apps/design/backend/src/app.ts [hunk 1]
```diff
@@ -304,52 +307,84 @@ export function buildApi(ctx: AppContext) {
...
-              candidate.middleName !== undefined &&
-              candidate.lastName !== undefined
-            ) {
-              return candidate;
-            }
-            return {
-              ...candidate,
-              ...splitCandidateName(candidate.name),
+      const electionResult = ((): Result<Election, Error> => {
+        switch (input.upload.format) {
```
Comment (kofi-q) [inline apps/design/backend/src/app.ts]: The reused common code paths in this method are so minimal that I think could be clearer to have explicit methods like `loadElectionVxf()`, `loadElectionMsSems()`, etc (or something along those lines), since we know which path we're taking at request-time on the client anyway. Wary of the potential road to large switch-case method with additional special-case fields in the input. WDYT?

Comment (jonahkagan) [reply apps/design/backend/src/app.ts]: I'll mock up the change and see how it feels

Comment (jonahkagan) [reply apps/design/backend/src/app.ts]: Ok, here's a sketch of what it looks like: https://github.com/votingworks/vxsuite/pull/7405/files. I think the server side does end up a bit clearer, but the frontend is a bit hairy, mainly because of way the form is weirdly split across two parts of a modal. The selector for which file format to use is part of the modal contents, but the action buttons are a separate prop to the modal. I would like to just separate out the forms into two separate components with separate form state, but I can't think of a simple way to do that. I could swap out the entire modal, but then both modals would need to have the format selector within them. Curious if you have any ideas here or any new thoughts after seeing what this code looks like

File: apps/design/backend/src/convert_ms_election.ts [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+          candidates,
+        };
+      }
+
+      // Ballot measure
+      case '1': {
+        const [, , contestLabel] = row;
+        const [titlePrefix, ...descriptionLines] = contestText.split(/\\n/);
+        const contestTitle = `${titlePrefix}: ${contestLabel}`;
+        const description = descriptionLines.join('\n').trim();
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply apps/design/backend/src/convert_ms_election.ts]: Oh great callout, will look into this, thanks
